app/Dao/UserDao.php

```diff
@@ -0,0 +1,75 @@
+<?php
+namespace App\Dao;
+
+use App\Contracts\Dao\UserDaoInterface;
+use App\Models\User;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Collection;
+use Illuminate\Http\Request;
+
+class UserDao implements UserDaoInterface
+{
+    /**
+     * Insert Data
+     *
+     * @param array $insertData
+     * @return void
+     */
+    public function insert(array $insertData): void
+    {
+        User::create($insertData);
+    }
+
+    /**
+     * Get Users By Id
+     *
+     * @param int $id
+     * @return User
+     */
+    public function getUserById(int $id): User
+    {
+        $data = User::find($id);
+        return $data;
+    }
+
+    /**
+     * Delete User
+     *
+     * @param int $id
+     * @return void
+     */
+    public function delete(int $id): void
+    {
+        User::whereId($id)->delete();
+    }
+
+    /**
+     * Update User
+     *
+     * @param array $updateData
+     * @param integer $id
+     * @return void
+     */
+    public function update(array $updateData, int $id): void
+    {
+        $user = User::find($id);
+        $user->name = $updateData['name'];
+        $user->email = $updateData['email'];
+        $user->role = $updateData['role'];
+        if ($updateData['img']) {
+            $user->img = $updateData['img'];
+        }
+        $user->save();
```

Comment:
You can use just single code line for this function. 
`User::where('id', $id)->update($updateData);`